File: 1823-find-the-winner-of-the-circular-game/1823-find-the-winner-of-the-circular-game.java
class Solution {
    public int findTheWinner(int n, int k) {
        
        ArrayList<Integer> list = new ArrayList<>();
        int del_pos=0;
        
        for(int i=0; i<n; i++){
            list.add(i+1);
        }
        
        while(list.size()!=1){
            del_pos = (del_pos+k-1)%list.size();
            list.remove(del_pos);
        }
        return list.get(0);
    }
}

File: 1823-find-the-winner-of-the-circular-game/1823-find-the-winner-of-the-circular-game.cpp
class Solution {
public:
    int answer(vector<int> &vec, int i, int k){
        
        if(vec.size()==1)
            return vec[0];
        
        int del_pos = (i+k-1)%vec.size();
        vec.erase(vec.begin()+del_pos);
        
         return answer(vec, del_pos, k);
    }
    int findTheWinner(int n, int k) {
        
        vector<int> v(n);
        
        for(int i=0; i<n; i++){
            v[i] = i+1;
        }
        
       return answer(v, 0, k);
        
    }
    
};